src/middleware/immer.ts

```diff
@@ -57,9 +57,19 @@ const immerImpl: ImmerImpl = (initializer) => (set, get, store) => {
   type T = ReturnType<typeof initializer>
 
   store.setState = (updater, replace, ...a) => {
```

Comment:
Let's remove `shouldReplace?: boolean | undefined` on line `45` and remove `replace` parameter since wouldn't be necessary

```suggestion
  store.setState = (updater, ...a) => {
```

Reply:
Shouldn't we also remove `...a` and maybe change `nextStateOrUpdater: T | Partial<T> | ((state: Draft<T>) => void)` to `updater: (state: Draft<T>) => void`.

I'll be honest I'm very confused with the typing in these middlewares 😅